nixos/modules/services/networking/nixops-dns.nix

```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  pkg = pkgs.nixops-dns;
+  cfg = config.services.nixops-dns;
+in
+
+{
+  options = {
+    services.nixops-dns = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the nixops-dns resolution
+          of NixOps virtual machines via dnsmasq and fake domain name.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        description = ''
+          The user the nixops-dns daemon should run as.
+        '';
+      };
+
+      domain = mkOption {
+        type = types.str;
+        description = ''
+          Fake domain name to resolve to NixOps virtual machines.
+
+          For example "ops" will resolve "vm.ops".
+        '';
+        example = "ops";
+        default = "ops";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.nixops-dns = {
+      description = "nixops-dns: DNS server for resolving NixOps machines";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        ExecStart="${pkg}/bin/nixops-dns --domain=.${cfg.domain}";
+      };
+    };
+
+    services.dnsmasq = {
+      enable = true;
+      resolveLocalQueries = true;
+      servers = [
+        "/${cfg.domain}/127.0.0.1#5300"
+      ];
+      extraConfig = ''
+        bind-interfaces
+        listen-address=127.0.0.1
```

Comment:
Would it not make more sense to set `listenAddress` to port 53 and give the service `cap_net_bind_service` capability to bind privileged ports? You can take a look at the caddy service on how to do this.

Reply:
To be able to still resolve internet domains via dnsmasq while only `*.ops` are forwarded to `nixops-dns`

Reply:
There should be an option to disable dnsmasq in the nixops-dns module to make it less opinionated. Also it is a sane default some other people maybe prefer to run `unbound` or `knot-resolver` instead.

Reply:
@Mic92 do you consider this a blocker or can it be added/iterated upon later?

Reply:
I'll update this to make dnsmasq setup an option (enabled by default).

Reply:
Done.